lib/logstash/inputs/udp.rb

```diff
@@ -97,10 +101,12 @@ def inputworker(number)
           decorate(event)
           event.set("host", client[3]) if event.get("host").nil?
           @output_queue.push(event)
+          metric.increment(:events)
```

Comment:
After taking the chance to investigate the logstash code where actually codecs are being created, see https://github.com/elastic/logstash/blob/master/logstash-core/lib/logstash/config/mixin.rb#L379-L410 for more details where actually codecs are being instanciated.

The only option to implement this in a nice way would be to properly "inject" the metrics object into the codec, but the logic as of now is all coupled with validating all the parameters, see https://github.com/elastic/logstash/blob/master/logstash-core/lib/logstash/config/mixin.rb#L452 for exact details where codecs are right now created. As of now, without a bit of refactoring I don't see a nice was, as nice as https://github.com/elastic/logstash/blob/master/logstash-core/lib/logstash/pipeline.rb#L401-L405, to sort out the "injection".

For this I do prefer that for now, we avoid having this in the codecs. we can always workout a proper solution when things like https://github.com/elastic/logstash/issues/4765 landed and we clean up this area.